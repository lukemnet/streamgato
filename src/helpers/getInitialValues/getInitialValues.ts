interface FormField {
  [key:string]: unknown;
  value: string | number;
}

interface FormParam {
  [key:string]: FormField;
}

interface InitialValues {
  [key:string]: string | number;
}

const getInitialValues = (params: FormParam) => {
  const fieldNames = Object.keys(params);
  let initialValues = {} as InitialValues;

  fieldNames.forEach(fieldName => {
    initialValues[fieldName] = params[fieldName].value;
  });

  return initialValues;
}

export default getInitialValues;
