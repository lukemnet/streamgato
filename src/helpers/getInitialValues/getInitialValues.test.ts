import getInitialValues from './getInitialValues';

const testParams = {
  hours: {
    type: "number",
    label: "Hours",
    shorthand: "h",
    value: 1,
  },
  minutes: {
    type: "number",
    label: "Minutes",
    shorthand: "m",
    value: 5,
  },
  seconds: {
    type: "number",
    label: "Seconds",
    shorthand: "s",
    value: 30,
  },
}

it('should return initial values', () => {
  expect(getInitialValues(testParams)).toMatchSnapshot();
})