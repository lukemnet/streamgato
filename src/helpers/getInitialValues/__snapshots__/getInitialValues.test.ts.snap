// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return initial values 1`] = `
Object {
  "hours": 1,
  "minutes": 5,
  "seconds": 30,
}
`;
